doc/develop/west/build-flash-debug.rst

```diff
@@ -386,6 +386,15 @@ You can :ref:`configure <west-config-cmd>` ``west build`` using these options.
            to the source directory. If the current working directory is inside
            the source directory this will be set to an empty string.
          - ``app``: The name of the source directory.
+   * - ``build.gdb``
+     - String. If present and nonempty, this is the path to the GDB executable
+       that will be used by default if you use commands like ``west debug`` and
+       ``west attach`` on the generated build directory. For example, you can
+       set this to the path to the ``arm-zephyr-eabi-gdb-py`` executable in the
+       :ref:`toolchain_zephyr_sdk` to debug Arm Cortex-M applications with
+       Python available from within GDB by default. Be careful with this option
+       if you switch architectures often, as GDB binaries for different target
+       architectures than the board you are currently debugging will not work.
```

Comment:
Now that makes me wonder if we should be shipping `zephyr-gdb-multiarch` in the SDK ...

Reply:
> Now that makes me wonder if we should be shipping zephyr-gdb-multiarch in the SDK ...

Uh, is that an option? If so, that sounds lovely.

Reply:
Yes, GDB, unlike other GNU toolchain components, supports "multi-arch" build (i.e. multiple target architecture support in single GDB executable).

I have filed https://github.com/zephyrproject-rtos/sdk-ng/issues/581.

Reply:
For my use-case, using "gdb-py" instead of "gdb" is all I wanted to change.

It appears that I had my issue because I was using SDK 0.15.0 without the zephyr changes in (i.e before last NCS upmerge of Zephyr).

With 3363c64abc6887a8ec40c1e59d9e4a3acbb21d00 I do get the gdb variant that I want.

The option introduced here could still be useful, but maybe an zephyr GDB-prefix option could be added that would make this easier for other toolchains than the Zephyr SDK, or for users that by default don't want the python cabable GDB executable.

It gets confusing with the gdb, gdb-py gdb-no-py variants.